hbase-diagnostics/src/test/java/org/apache/hadoop/hbase/wal/TestBoundedRegionGroupingStrategy.java

```diff
@@ -80,7 +75,7 @@ public static Iterable<Object[]> data() {
 
   @Before
   public void setUp() throws Exception {
-    CONF.set(DELEGATE_PROVIDER, walProvider);
+    CONF.set(RegionGroupingProvider.DELEGATE_PROVIDER, walProvider);
```

Comment:
Sure let me have a look at these

Reply:
Please see https://github.com/apache/hbase/pull/6258#discussion_r1781580248

Reply:
WALPerformanceEvaluation is in the main (non-test) hbase-diagnostics JAR.

Wouldn't adding a test scoped dependency on hbase-diagnostics to HBase server fix the issue ?
I proposed allowing hbase-diagnostics as a test scope dependency (either main jar, or even the test jar) in other modules to allow for cases like this.

IMO we're fine as long as there is no compile/runtime scopt dependency on hbase-diagnostics.

Reply:
hbase-diagnostics test code depends on hbase-server test jar for building. it uses HBaseTestingUtility. 

If I understood you correctly, we are proposing to add hbase-diagnistics as test dependency to hbase-server tests, but this will cause cyclic dependency.

Currently hbase-diagnostics builds only after hbase-server and hbase-mapreduce are built.

Reply:
Thanks for clarifying.

So it's either leaving WALPerformanceEvaluation in HBase-server, or moving the dependent tests to hbase-diagnostics.

Reply:
> hbase-diagnostics test code depends on hbase-server test jar for building. it uses HBaseTestingUtility.

This detail I missed. I was thinking that the purpose of `hbase-diagnostics` was to be something that shipped without any test dependencies. Is that structure not practical?

Reply:
IMO it' fine for the tests in hbase-diagnostics (i.e tests for the diagnostics tools) to depend on test jars from other modules, as long as the non-test jar (i.e. the diagnostics tools themselves) don't depend on test jars.

This way we can add the diagnostics tools to the assembly without having to add any test jars or dependencies.

Reply:
Yes, that's my understanding. I was expecting that the hbase-diagnostics.jar would have no dependencies on -test.jars, while the hbase-diagnostics-test:jar would have dependencies on other -test.jars.

I believe that we are agreed, @stoty . I misunderstood @NihalJain 's earlier comment? Thanks for clarifying.